sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -294,7 +299,8 @@ public long getEstimatedSizeBytes(PipelineOptions pipelineOptions) {
      * @param additionalFilter A custom (user) additional filter to append to the range filters.
      * @return A list of filters containing the ranges.
      */
-    private static List<String> splitKeysToFilters(List<Document> splitKeys, String
+    @VisibleForTesting
+    protected static List<String> splitKeysToFilters(List<Document> splitKeys, String
```

Comment:
optional: Add a precondition that checks that splitKeys is not empty since this method does not correctly handle the empty splitKeys case.